alpa/pipeline_parallel/stage_profiling.py

```diff
@@ -428,27 +426,22 @@ def __init__(self, num_cpus, num_gpus, prof_result, mesh_num_devices,
         self.pool = ActorPool(self.actors)
 
 
-def compile_all(stages):
+def compile_all(stages, num_micro_batches, default_as_option):
     """
     Compile all input stages.
-
-    Args:
-        stages: List of info for compilation.
     """
     num_cpus = int(
         min(max(ray.available_resources()["CPU"] // 2, 1), len(stages)))
-    num_gpus = int(ray.available_resources()["GPU"])
-    default_autosharding_option = global_config.default_autosharding_option
 
-    compile_workers = CompileWorkerPool(num_cpus, num_gpus)
+    compile_workers = CompileWorkerPool(num_cpus)
     for stage_id, (_, stage_config, auto_sharding_config,
                    _) in enumerate(stages):
         logical_mesh, autosharding_option_dict = auto_sharding_config
         compile_workers.submit(
             lambda w, v: w.compile_stage_for_profiling.remote(*v),
             (stage_id, stage_config.compile_config, logical_mesh,
-             default_autosharding_option.deepcopy_and_update(
-                 autosharding_option_dict), global_config.num_micro_batches))
+             dataclasses.replace(default_as_option, **autosharding_option_dict),
```

Comment:
not critical to this PR. Can we use an `update` method just like what in previous GlobalConfig?

Reply:
Using immutable copies is prefered especially the `default_as_option` is used multiple times